Functions/Assertions/Should.ps1

```diff
@@ -53,17 +53,18 @@ function Should {
 
     begin {
         $inputArray = [System.Collections.Generic.List[PSObject]]@()
+        $lineNumber = $MyInvocation.ScriptLineNumber
+        $lineText = $MyInvocation.Line.TrimEnd("$([System.Environment]::NewLine)")
+        $file = $MyInvocation.ScriptName
     }
 
     process {
         $inputArray.Add($ActualValue)
+
+        $ActualValue
```

Comment:
Returning the input value is probably not what we always want, is it? For example when doing `Should -Throw` will that assertion return the exception that was caught and not the input values?

Reply:
Good point. But should we consider `Should -Throw` a one-off scenario? Because the idea is to pass the original input to subsequent Should cmdlets, which in this case would be a `scriptblock`.

Or do you prefer to include a check for `Should -Throw`? I believe this will require us to hightlight it somehow in the documentation

Reply:
The scriptblock does not matter on the output, at least not as much as the exception. Getting the exception from the failed scriptblock is way more practical and interesting than getting the scriptblock. So yes Throw is an exception. Should Throw should return the error record, and Should -Not Throw should return nothing. IMO

Reply:
This would be another change of behavior. Currently the error record is sent in the pipeline if the `PassThru` switch is specified. Do you want me to remove the PassThru parameter and return the error record everytime?

Reply:
Yes please, otherwise the assertion chaining would really suck. You'd have to specify -PassThru all the time.